### PR TITLE
Add XACML service for UMA 2.0 component.

### PR DESCRIPTION
**Pull request by isuri97**
This PR represents the how policy evaluation performs on  UMA 2.0. In this scenario, as the policy defining language, XACML is used and there can be more ways to implement this part to UMA 2.0  as a pluggable extension.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/test/resources/testng.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ~ limitations under the License.
+  -->
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="Identity-Oauth-Uma-Endpoint-Test-Suite"
+       object-factory="org.powermock.modules.testng.PowerMockObjectFactory">
+    <test name="Util-Tests" preserve-order="true" parallel="false">
+        <classes>
+            <class name="org.wso2.carbon.identity.oauth.uma.xacml.service.handler.XACMLUMAHandler" />
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/test/resources/testng.xml]: Is this a test class?

File: components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/test/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.wso2.carbon.identity.oauth.uma.xacml.service.handler;
+
+
+/**
+ * Created by isuri on 5/30/18.
+ */
+public class XACMLUMAHandlerTest {
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/test/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandlerTest.java]: Why we are having an empty test class without any methods.

Comment (mefarazath) [reply components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/test/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandlerTest.java]: Also setup your IDE to include license and to remove default class level comment

File: components/org.wso2.carbon.identity.oauth.uma.xacml.service/pom.xml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-rs-service-description</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.core</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <!--Test Dependencies-->
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth.uma.xacml.service/pom.xml]: As there is no Test case implemented. We can remove these dependencies

File: components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    /**
+     * @param userName
+     * @param resource
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    public boolean isAuthorized(String userName, List<Resource> resource) throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("In policy authorization flow..." + "User Name" + " " + userName);
```
[the remark this answers is not in the document]
Comment (isuri97) [reply components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java]: Done.

File: components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        }
+        return isValid;
+    }
+
+    private RequestDTO createRequestDTO(Resource resource, String userName) {
+
+        String listString = "";
+        List<RowDTO> rowDTOs = new ArrayList<>();
+        for (String scopes : resource.getResourceScopes()) {
+            listString += scopes + ",";
```
[the remark this answers is not in the document]
Comment (isuri97) [reply components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java]: Done.

File: components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+                String requestString = PolicyBuilder.getInstance().buildRequest(requestElementDTO);
+                if (log.isDebugEnabled()) {
+                    log.debug("XACML Authorization request :\n" + requestString);
+                }
+
+                String responseString =
+                        AppUMADataholder.getInstance().getEntitlementService().getDecision(requestString);
+                if (log.isDebugEnabled()) {
+                    log.debug("XACML Authorization response :\n" + responseString + "User Name" + " " + userName);
```
[the remark this answers is not in the document]
Comment (isuri97) [reply components/org.wso2.carbon.identity.oauth.uma.xacml.service/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java]: Done

File: components/org.wso2.carbon.identity.oauth.uma.xacml.extention/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+public class XACMLUMAHandler implements PolicyEvaluator {
+
+    private static final Log log = LogFactory.getLog(XACMLUMAHandler.class);
+    private static final String DECISION_XPATH = "//ns:Result/ns:Decision/text()";
+    private static final String XACML_NS = "urn:oasis:names:tc:xacml:3.0:core:schema:wd-17";
+    private static final String XACML_NS_PREFIX = "ns";
+    private static final String RULE_EFFECT_PERMIT = "Permit";
+    private static final String RULE_EFFECT_NOT_APPLICABLE = "NotApplicable";
+
+    public XACMLUMAHandler() {
```
Comment (nuwandiw) [inline components/org.wso2.carbon.identity.oauth.uma.xacml.extention/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/handler/XACMLUMAHandler.java]: Shall we remove the public constructor?

File: components/org.wso2.carbon.identity.oauth.uma.xacml.extention/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/internal/AppUMADataholder.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package org.wso2.carbon.identity.oauth.uma.xacml.service.internal;
```
Comment (nuwandiw) [inline components/org.wso2.carbon.identity.oauth.uma.xacml.extention/src/main/java/org/wso2/carbon/identity/oauth/uma/xacml/service/internal/AppUMADataholder.java]: Move package name below the license header

**Pull request merged**